src/reader.rs

```diff
@@ -20,6 +20,30 @@ use std::convert::From;
 use std::io::{self, Read};
 use std::iter;
 
+/// Trait to allow generic objects (not just BTreeMap) in some methods.
+pub trait StoreObjs {
+    /// Insert given object at given key index.
+    fn insert(&mut self, key: OsmId, value: OsmObj);
+    /// Check if object contains the given key.
+    fn contains_key(&self, key: &OsmId) -> bool;
+    /// Check object corresponding to the given key.
+    fn get(&self, key: &OsmId) -> Option<OsmObjWrapper>;
```

Comment:
I think you don't need this OsmObjWrapper, but you want something like that:
```rust
pub trait StoreObjs {
    type Ref: std::borrow::Borrow<OsmObj>;
    fn get(&self, key: &OsmId) -> Option<Self::Ref>;
    // ...
}
```
```

Reply:
Looking at your usecase, you don't need dynamic ref/value segregation, you need static, thus this proposed way seems better.

Reply:
No because then I can't use methods of the object or I'd need to implement more traits. I tried as well and it didn't work well at all.

Reply:
```rust
pub trait StoreObjs {
    type Ref: std::ops::Deref<OsmObj>;
    fn get(&self, key: &OsmId) -> Option<Self::Ref>;
    // ...
}
```
then?

Reply:
The `get` method is not needed by the function, so you can just remove it from the trait, and do whatever you want on the user side to get back the objects.

Reply:
I'm not sure deref is implemented for T (but maybe I'm completely wrong on this one, need to check!)... And since the sql implementation cannot return a reference...

Reply:
Hum... It means then that I'll have to create another trait on mimir side but it can be done. If you prefer it that way?

Reply:
No will not work with `Deref`, but as `get` is not needed, I doesn't need to be in the trait, and the user can do whatever he want.

Reply:
I'd prefer it this way yes.

Reply:
Removed the get method.